### DISCUSSION TITLE
`plugin.py` conflict

### DISCUSSION THREAD

**Discussion by eirnym** [pytest-dev/pytest#12724] [General] [answered]
I'm writing a plugin for a certain system and specification directly tells to use `plugin.py` as an entry point. I have problems when I testing using `pytest` and wonder how to exclude this file from pytest's automatic plugin discovery

My directory layout looks like:

```
src/plugin.py
tests/
    conftest.py
    test_functionality.py
    test_other_functionality.py
```

**Comment by RonnyPfannschmidt** [comment]
More context is needed, pytest doesn't consider random plugin.py files as plugin

**Reply by eirnym** [reply]
with pytest `7.4.1` I get results below if I try to add `src` to `PYTHONPATH` (via environment variable or pytest.ini) or even just creating `pytest.ini` file. Without these settings I obviously can't find my `plugin.py` to import some to test functions from.

I've tried to recreate venv and have exactly same results

```
/Users/user/Developer/project/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:330: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: helpconfig, Hook: pytest_cmdline_parse
ConftestImportFailure: ImportError: cannot import name 'BackendMode' from 'plugin' (/Users/user/project/src/plugin.py) (from /Users/user/project/tests/tests_steam_network/conftest.py)
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config = pluginmanager.hook.pytest_cmdline_parse(
ImportError while loading conftest '/Users/user/project/tests/tests_steam_network/conftest.py'.
tests/tests_steam_network/conftest.py:6: in <module>
    from plugin import BackendMode
E   ImportError: cannot import name 'BackendMode' from 'plugin' (/Users/user/project/src/plugin.py)
```

**Reply by RonnyPfannschmidt** [reply, marked as answer]
Src layout without editable install needs a python path addition

**Reply by eirnym** [reply]
Thank you for your time, I made it work, but output with pluggy and old style teardown made me worry.

**Reply by shishir20may2022** [reply]
what is the resolution

**Reply by RonnyPfannschmidt** [reply]
In this case adding src to the pythonpath 

I Generally suggest editable installs
